sql/src/main/java/org/apache/druid/sql/calcite/rel/VirtualColumnRegistry.java

```diff
@@ -136,9 +137,11 @@ public RowSignature getFullRowSignature()
     final RowSignature.Builder builder =
         RowSignature.builder().addAll(baseRowSignature);
 
+    ColumnInspector baseRowsInspector = builder.build().asColumnInspector();
```

Comment:
Does this mean a virtual column can't reference another virtual column? since the ColumnInspector is built with just he `baseRowSignature`?

Reply:
yeah, but this was true before. It would be possible to examine the expressions and resolve an execution order probably to allow it, but we haven't built it yet. This code would need to change to match I think